packages/go_router/lib/src/misc/extensions.dart

```diff
@@ -89,4 +99,29 @@ extension GoRouterHelper on BuildContext {
         queryParams: queryParams,
         extra: extra,
       );
+
+  /// Replaces the top-most page of the page stack with the given one.
```

Comment:
Can you also mention the transition and preserve state behavior somewhere in here as well?